design-philosophy-and-threat-model.md

```diff
@@ -0,0 +1,128 @@
+    - defense in depth, 
+    - graceful failures, 
+    - distribution of trust, no single point of failure, 
+    - least privilege, 
+    - minimize attack surface
+- Acknowledgment of human fallibility, importance of usability
+- Trust-but-verify, open source, abort & identify malicious behavior
+- Consideration of challenges to security posed by integration with the service provider’s asset management product: 
+    - Our software should be conducive to integration in a way that protects against threats not in our immediate scope; 
+    - The importance of providing clear, actionable guidance for organizational partners
+- Extensibility, modularity 
+    - Including allowing for transition to more robust threat models, e.g., fault-tolerant Byzantine servers
+    - Generalizability
+    - Keep components modular so that we can serve different types of community members and service providers with different constraints/end goals. (e.g., Do not preclude asset owner from running their own server; allow asset owners to choose from a menu of custodial options; allow service providers to choose from a menu of deployment options)
+- Deployment flexibility:
+    - Support on-premises HSMs and in-cloud servers
+    - Allow use of external cloud infrastructure while minimizing trust required by using secure processors and/or secure hardware.
+    - Allow distribution of trust by using multiple geographic/legal jurisdictions and/or external cloud/service providers
+
+## Threat Model and Mitigations
+
+The system should be secure against both external and internal actors, as detailed below. We ensure compatitibiliy of our technology with mitigations against the following types of adversaries and attacks. Many of the appropriate mitigations are outside the implementation scope of our core technology, but we advise the Service Provider to consider the following types of attacks and mitigation measures when deploying a digital asset manager built with Lock Keeper.
+
+A core principle in the development of Lock Keeper is to provide guidance and make decisions that enable the consuming application to create a secure and usable product.
+
+[External Actors](#external-actors)<br>
+[Internal Actors](#internal-actors) <br>
+[Software Dependencies](#software-dependencies)<br>
+[Additional Threat Considerations](#additional-threat-considerations)
+
+### External actors
+
+#### Passive and active network-level adversaries
+We advise consideration of the following:
+
+1. Denial of service (DoS) attacks, including both temporary and permanent (e.g., volumetric, loss of server): For deployment,  standard DoS mitigations at both the cryptographic and the networking level are advisable, including mechanisms that account for server compromise and failure, namely
+    1. Server over-provisioning; 
+    1. Server replication; 
+    1. Automated volumetric DoS protection; and
+    1. Cold backups, where/if appropriate.
```

Comment:
```suggestion
    1. Server over-provisioning 
    1. Server replication 
    1. Automated volumetric DoS protection
    1. Cold backups, where/if appropriate
```

Reply:
I don't intend to leave off punctuation in lists in general. The "philosophy" section needs to be expanded as part of a separate ticket. Planning to write the ticket after merging this PR.